### PR TITLE
Move websocket-util classes into websocket-core

### PR DESCRIPTION
**Pull request by lachlan-roberts**
- Move the `websocket-util` classes into `websocket-core-common` internal package.
- Move the `websocket-util-server` internal classes into the `websocket-core-server` API.
- API cleanups of websocket-core
- `WebSocketUpgradeHandler` now uses the `WebSocketMappings` and can upgrade to multiple different `FrameHandler`s.
- `WebSocketUpgradeFilter` now exists in a jar all by itself.

### CODE REVIEW

**Comment by lachlan-roberts** [comment]
@gregw @sbordet There is still something strange about the websocket structure. The `websocket-jetty-common` module is really just a private implementation but it leaks a bunch of websocket-core classes in its method signatures. It is used as common implementation between `websocket-jetty-server` and `websocket-jetty-client`, you are never expected to directly use classes from `websocket-jetty-common`. This is different to  `websocket-core-common`, you need to use classes directly from this jar as there is no `websocket-core-api`.

And there is JPMS warnings because these internal classes from `websocket-core-common` are in the public signatures of `websocket-jetty-common` and `websocket-javax-common`, even though these are really just meant to be internal implementation jars.

**Comment by sbordet** [comment]
@lachlan-roberts there are a bunch of test failures due to the latest changes.

The reason is that now JPMS module `org.eclipse.jetty.websocket.jetty.common` is only exported to `org.eclipse.jetty.websocket.jetty.client` and `*-server`.

However, there are test classes that are in the `org.eclipse.jetty.websocket.jetty.common` module, and for tests that's the _application_ `Lookup` object.

However, classes from that package/module are not exported so the `Lookup` object has `allowedModes=0` (which means no permissions), which means that we get an `IllegalAccessException` when trying to call `findVirtual()` to find the `@OnMessage` method.

**Comment by sbordet** [comment]
Feels like a JDK bug... I sent an email to jigsaw-dev asking for clarifications.

**Comment by sbordet** [comment]
@lachlan-roberts jigsaw-dev replied and classes from `org.eclipse.jetty.websocket.jetty.common` must be exported _unconditionally_.

I'm making this change and see if the PR builds fine.

File: Jenkinsfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ pipeline {
           steps {
             container('jetty-build') {
               timeout( time: 120, unit: 'MINUTES' ) {
```
Comment (joakime) [inline Jenkinsfile]: You missed the timeout in the merge.

Comment (olamy) [reply Jenkinsfile]: no because I'm interested to have time log into the pod and get the logs :)

Comment (olamy) [reply Jenkinsfile]: oh you mean longer timeout?

Comment (olamy) [reply Jenkinsfile]: arfff :)

Comment (olamy) [reply Jenkinsfile]: arf got it

File: tests/test-integration/pom.xml [hunk 2]
```diff
@@ -176,5 +177,10 @@
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <version>1.2.3</version>
```
Comment (joakime) [inline tests/test-integration/pom.xml]: Wouldn't this conflict with jetty-slf4j-impl ?

Comment (olamy) [reply tests/test-integration/pom.xml]: Oh yeah I agree but I removed my tests from this branch

**Review by sbordet — CHANGES_REQUESTED** [review]
Few nibble to fix.

**Review by sbordet — APPROVED** [review]

**Pull request merged**